luxjs/google/module.js

```diff
@@ -55,20 +81,32 @@
                                 scrollwheel: config.scrollwheel ? true : false,
                             },
                             map = new google.maps.Map(element[0], opts),
-                            //
                             marker = config.marker;
+                        //
+                        self.addMarker(map, marker, loc);
 
-                        if (marker)
-                            var gmarker = new google.maps.Marker(angular.extend({
-                                position: loc,
-                                map: map,
-                            }, marker));
+                        // Allow different directives to use this map
+                        scope.map = map;
                         //
                         windowResize(function () {
                             google.maps.event.trigger(map, 'resize');
                             map.setCenter(loc);
                             map.setZoom(map.getZoom());
                         }, 500);
+                    };
+                },
+                //
+                link: function (scope, element, attrs, controller) {
+                    if(!scope.googlemaps) {
+                        $lux.log.error('Google maps url not available. Cannot load google maps directive');
+                        return;
+                    }
+                    require([scope.googlemaps], function () {
+                        controller.initialize(scope, element, attrs);
+
+                        // Setup map by another directive
+                        if (scope.hasOwnProperty('setup'))
```

Comment:
Are you just checking that scope.setup() is defined? If so then it's better to say
```if (typeof scope.setup === 'function')```
Or do you actually need to make sure you are using this particular scope's setup method, and do not want to use an inherited one? That seems unlikely, but I am not familiar with this code.